fvm/evm/emulator/state/stateDB.go

```diff
@@ -308,12 +308,44 @@ func (db *StateDB) SetTransientState(addr gethCommon.Address, key, value gethCom
 
 // AddressInAccessList checks if an address is in the access list
 func (db *StateDB) AddressInAccessList(addr gethCommon.Address) bool {
-	return db.latestView().AddressInAccessList(addr)
+	// For each static call / call / delegate call, the EVM will create
+	// a snapshot, so that it can revert to it in case of execution errors,
+	// such as out of gas etc, using `Snapshot` & `RevertToSnapshot`.
+	// This can create a long list of views, in the order of 4K for certain
+	// large transactions. To avoid performance issues with deep stacks,
+	// we use a plain for-loop, instead of checking parents, as DeltaView did.
```

Comment:
```suggestion
	// large transactions. To avoid performance issues with DeltaView checking parents,
	// which causes deep stacks and function call overhead, we use a plain for-loop instead.
```